bazel/js/internals/module.bzl

```diff
@@ -0,0 +1,40 @@
+load("@build_bazel_rules_nodejs//:index.bzl", "js_library")
+load("@npm//@bazel/typescript:index.bzl", "ts_project")
+
+# Defines this as an importable module area for shared macros and configs
+
+def build_module(package_name, srcs, deps, **kwargs):
+    """
+    Macro that allows easy composition of routes from a multi route spa
+
+    Args:
+        package_name: name of the package
+        srcs: source files from the package
+        data: any dependencies needed to build
+        kwargs: any other inputs to js_library
+    """
+
+    # list of all transpilation targets from SWC to be passed to webpack
+    ts_project(
+        name = "transpiled_pkg",
+        srcs = srcs,
+        deps = deps,
+        declaration = True,
+        tsconfig = {
+            "compilerOptions": {
+                "declaration": True,
+            },
+        },
+    )
+
+    visibility = kwargs.pop("visibility", ["//visibility:public"])
```

Comment:
Public is definitely overly broad. Would rather folks explicitly declare what they want instead of us just granting it